lib/rex/proto/ldap/server.rb

```diff
@@ -149,104 +155,177 @@ def dispatch_request(cli, data)
         #
         # Default LDAP request dispatcher
         #
-        # @param cli [Rex::Socket::Tcp, Rex::Socket::Udp] Client sending the request
+        # @param client [Rex::Socket::Tcp, Rex::Socket::Udp] Client sending the request
         # @param data [String] raw LDAP request data
-        def default_dispatch_request(cli, data)
-          return if data.strip.empty?
+        def default_dispatch_request(client, data)
+          return if data.strip.empty? || data.strip.nil?
+
+          state = {}
+          post_pdu = false
+          processed_pdu_data = {}
+          state[client] = {
+            name: "#{client.peerhost}:#{client.peerport}",
+            ip: client.peerhost,
+            port: client.peerport,
+            service_name: 'ldap'
+          }
```

Comment:
The `state` hash here is not consumed as a hash in this refactor consider just using a single layer hash to hold the details of the client when you need to pass it on.  Really since this is only accessed once in the method you could just generated the hash when you need to merge it to the processed data.

```suggestion
          post_pdu = false
          processed_pdu_data = {}
          client_details = {
            name: "#{client.peerhost}:#{client.peerport}",
            ip: client.peerhost,
            port: client.peerport,
            service_name: 'ldap'
          }
```